### PR TITLE
fix(vrl): ignore comments in nested code

### LINKED ISSUE #8326: VRL comment inside of `if` block with `'` causes a syntax error
### Vector Version

```
vector 0.14.0 (x86_64-unknown-linux-gnu 5f3a319 2021-06-03)
```

### Vector Configuration File

VRL snippet from a remap source.

```
if contains!(.message, "bytes_rx=") {
  # it's a connection/request
  .log_type = "request"
}
```

### Expected Behavior

Valid VRL. An if with a comment inside the branch.

### Actual Behavior

```
vector_1   | error[E202]: syntax error
vector_1   |   ┌─ :1:1
vector_1   |   │
vector_1   | 1 │ ╭ if contains!(.message, "bytes_rx=") {
vector_1   | 2 │ │   # it's a connection/request
vector_1   | 3 │ │   .log_type = "request"
vector_1   | 4 │ │ }
vector_1   | 5 │ │
vector_1   |   │ ╰^ unexpected error: invalid literal
vector_1   |   │
vector_1   |   = see language documentation at https://vrl.dev
vector_1   |
```

### Additional Context

If i remove the comment it compiles fine.

### PR DESCRIPTION
**Pull request by StephenWakely**
Closes #8326

The lexer was lexing code within comments within nested blocks. This would cause it to trip up if it encountered something that looked like an invalid string literal (things like `# It's not a timestamp`).

Signed-off-by: Stephen Wakely <fungus.humungus@gmail.com>

### CODE REVIEW

File: lib/vrl/parser/src/lex.rs [hunk 1]
```diff
@@ -1903,4 +1919,26 @@ mod test {
...
+
+    #[test]
+    fn comment_in_block() {
+        test(
+            data(indoc!(
+                r#"if x {
+                      # It's an apostrophe.
+                      3
+                   }"#
+            )),
```
Comment (pablosichert) [inline lib/vrl/parser/src/lex.rs:1931]: When reading the other test cases in this file, I noticed that they have been intentionally kept on a single line, such that the `~` in the assertion line up neatly, e.g. 

https://github.com/vectordotdev/vector/blob/b9c0b9581b0b4db6022192c0cadf721426b2570d/lib/vrl/parser/src/lex.rs#L1771-L1789

Not perfect, but I'd suggest doing the same here:

```suggestion
            data(r#"if x {\n# It's an apostrophe.\n3\n}"#)),
```

**Review by pablosichert — APPROVED** [review]
Looks good! Just one cosmetic suggestion.

File: lib/vrl/parser/src/lex.rs (hunk 1 above)
Comment (StephenWakely) [reply lib/vrl/parser/src/lex.rs]: Have made the change, but it kinda goes wrong anyway since the `\n` pushes everything out by one char.

Comment (pablosichert) [reply lib/vrl/parser/src/lex.rs]: Aaaah, I see... that's a shame. I wonder if we can fix that with reasonable effort. Would leave it like that for now, though.

**Pull request merged**